src/build.d

```diff
@@ -470,11 +470,12 @@ LtargetsLoop:
 
             default:
                 // check this last, target paths should be checked after predefined names
+                const tAbsolute = t.absolutePath.buildNormalizedPath;
                 foreach (dep; DependencyRange(rootDeps.map!(a => a()).array))
                 {
                     foreach (depTarget; dep.targets)
                     {
-                        if (depTarget.endsWith(t))
+                        if (depTarget.endsWith(t) || depTarget.endsWith(tAbsolute))
```

Comment:
`endsWith` is variadic: https://dlang.org/library/std/algorithm/searching/ends_with.html

Reply:
Incorporated this in https://github.com/dlang/dmd/pull/10203